libsolidity/formal/Why3Translator.cpp

```diff
@@ -77,12 +77,26 @@ string Why3Translator::toFormalType(Type const& _type) const
 			return "uint256";
 	}
 	else if (auto type = dynamic_cast<ArrayType const*>(&_type))
+	{
 		if (!type->isByteArray() && type->isDynamicallySized() && type->dataStoredIn(DataLocation::Memory))
 		{
 			string base = toFormalType(*type->baseType());
 			if (!base.empty())
 				return "array " + base;
 		}
+	}
+	else if (auto mappingType = dynamic_cast<MappingType const*>(&_type))
+	{
+		solAssert(mappingType->keyType(), "A mappingType misses a keyType.");
+		if (dynamic_cast<IntegerType const*>(&*mappingType->keyType()))
+		{
+			//@TODO Use the information from the key type and specify the length of the array as an invariant.
```

Comment:
If the `keyType` is `uint256`, then the length would be `2**256`, but when `keyType` is `address` the length would be `2**160`.  This would automatically add verification conditions that check balances are checked only for small numbers.